src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -90,26 +83,27 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
         }
 
         // dispatch to the correct handler
-        if (objResult.getsObjectType() == Allocation__c.sObjectType && !hasRunForAllocationsBefore && 
+        if (objResult.getsObjectType() == Allocation__c.sObjectType && !TDTM_ProcessControl.checkRecursionState(TDTM_ProcessControl.flag.ALLOC) &&
             (triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate)) {
```

Comment:
I've been doing some manual testing on allocations and it seems to be working well. I don't recall exactly why I split the allocations before/after flag, but I do remember having a single flag and running into some sort of bug that caused me to add the second one. It was definitely a reactive decision, not a proactive one. I think the new version is working as expected.